### PR TITLE
feat: ssh key auth

### PR DESCRIPTION
**Pull request by antmat**
Added ssh key authentication, key should be provided in sshd "AUTHORIZED_KEYS FILE FORMAT".
@3Hren @noxiouz PTAL

### CODE REVIEW

File: insonmnia/miner/server.go [hunk 1]
```diff
@@ -224,6 +231,15 @@ func (m *Miner) Start(ctx context.Context, request *pb.MinerStartRequest) (*pb.M
 		Resources:     transformResources(request.Resources),
 	}
 	log.G(m.ctx).Info("handling Start request", zap.Any("req", request))
+	var publicKey ssh.PublicKey = nil
```
Comment (noxiouz) [inline insonmnia/miner/server.go]: var ... is enough. no nil required, I think

File: insonmnia/miner/ssh.go [hunk 2]
```diff
@@ -55,8 +55,15 @@ func (s *sshServer) Run(miner *Miner) error {
 }
 
 func (s *sshServer) verify(ctx ssh.Context, key ssh.PublicKey) bool {
+	cinfo, ok := s.miner.GetContainerInfo(ctx.User())
+	if !ok {
+		return false
+	}
 	log.G(s.miner.ctx).Info("verifying public key")
-	return true
+	if ssh.KeysEqual(cinfo.PublicKey, key) {
```
Comment (noxiouz) [inline insonmnia/miner/ssh.go]: return ssh.KeysEqual?

Comment (antmat) [inline insonmnia/miner/ssh.go]: it was a long day today

Comment (noxiouz) [inline insonmnia/miner/ssh.go]: I know that feel, bro!

**Pull request merged**